enterprise/backend/test/metabase_enterprise/serialization/v2/e2e/yaml_test.clj

```diff
@@ -27,244 +25,335 @@
-              (is (= (clean-up-expected event)
-                     (yaml/from-file (io/file dump-dir "Timeline" timeline_id "TimelineEvent" filename))))))
+                                       count))))))
 
           (testing "for settings"
-            (is (= (into {} (for [{:keys [key value]} (get entities "Setting")]
-                              [key value]))
-                   (yaml/from-file (io/file dump-dir "settings.yaml"))))))
-
-        (testing "ingestion"
-          (let [ingestable (ingest.yaml/ingest-yaml dump-dir)]
-            (testing "ingest-list is accurate"
-              (is (= (into #{} (comp cat
-                                     (map (fn [entity]
-                                            (mapv #(cond-> %
-                                                     (:label %) (update :label #'u.yaml/truncate-label))
-                                                  (serdes.base/serdes-path entity)))))
-                           (vals entities))
-                     (into #{} (ingest/ingest-list ingestable)))))
-
-
-            (testing "each entity matches its in-memory original"
-              (doseq [entity extraction]
-                (let [->utc   #(t/zoned-date-time % (ZoneId/of "UTC"))]
-                  (is (= (cond-> entity
-                           true                                       (update :serdes/meta strip-labels)
-                           true                                       (dissoc :updated_at)
-                           ;; TIMESTAMP WITH TIME ZONE columns come out of the database as OffsetDateTime, but read back
-                           ;; from YAML as ZonedDateTimes; coerce the expected value to match.
-                           (t/offset-date-time? (:created_at entity)) (update :created_at ->utc))
-                         (ingest/ingest-one ingestable (serdes.base/serdes-path entity)))))))))))))
+            (is (.exists (io/file dump-dir "settings.yaml")))))
+
+          (testing "ingest and load"
+            (ts/with-dest-db
+              (testing "doing ingestion"
+                (is (serdes.load/load-metabase (ingest.yaml/ingest-yaml dump-dir))
+                    "successful"))
+
+              (testing "for Collections"
+                (doseq [{:keys [entity_id] :as coll} (get @entities "Collection")]
+                  (is (= (clean-entity coll)
+                         (->> (db/select-one 'Collection :entity_id entity_id)
+                              (serdes.base/extract-one "Collection" {})
+                              clean-entity)))))
+
+              (testing "for Databases"
+                (doseq [{:keys [name] :as coll} (get @entities "Database")]
+                  (is (= (clean-entity coll)
+                         (->> (db/select-one 'Database :name name)
+                              (serdes.base/extract-one "Database" {})
+                              clean-entity)))))
```

Comment:
It looks like there is a lot of repetition between each of these test parts... you might want to consider abstracting out the common parts into local functions with `let` or `letfn`. Maybe something like

```clj
(letfn [(test-model [model f]
          (testing (format "for %ss" (name model))
            (doseq [instance (get @entities (name model))]
              (f instance))))
        (test-extracted-entity [model instance-from-entities instance-from-app-db]
          (is (= (clean-entity instance-from-entities)
                 (-> instance-from-app-db
                     (serdes.base/extract-one (name model) {})
                     clean-entity))))]
  (test-model
   Table
   (fn [{:keys [db_id name], :as table}]
     (test-extracted-entity Table table (->> (db/select-one-field :id 'Database :name db_id)
                                             (db/select-one 'Table :name name :db_id))))))
```

could reduce the repetition a little bit. Idk